Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1376,9 +1370,29 @@ void MediaPlayerPrivateGStreamerBase::attemptToDecryptWithLocalInstance()
     if (is<CDMInstanceOpenCDM>(*m_cdmInstance)) {
         GST_DEBUG("handling OpenCDM %s keys", m_cdmInstance->keySystem().utf8().data());
         auto& cdmInstanceOpenCDM = downcast<CDMInstanceOpenCDM>(*m_cdmInstance);
-        String sessionId = cdmInstanceOpenCDM.getCurrentSessionId();
-        ASSERT(!sessionId.isEmpty());
-        dispatchDecryptionSession(sessionId);
+        size_t index = 0;
+        // Optimized this in case there's only one init data to avoid comparisons and because in some cases,
+        // the JS app send us down a new init data we can't map (in this case it would be a FIXME for the case
+        // it has more than one unmapped init datas).
```

Comment:
You have optimized but not too much since you still check for the count inside the loop, you are still running the loop. 

Anyway, taking again into consideration you comment:

> We have renamed m_initDataProtectionEventsMap to m_initDataProtectionEventsMapping. And logic changed to add only the protection event of selected KeySystem. Also events are appending to the same initData, now not creating new hash map. But still we cannot use m_initDataProtectionEventsMapping.size() for the purpose of m_initDataCount. Because this making issue in multi key case, i.e, after sending all events associated with an initdata, we are clearing corresponding initData map from the m_initDataProtectionEventsMapping. So once it reached at the last initData map case, it show .size() value as 1, so it will try to get current session instead of session id corresponding to initData.

Considering this, I think we should not avoid checking the init datas because of what you say, you can have more than one session left to handle but more than one in the CDM instance. We need to fix the algorithm then and always run the search in the init datas. Then, after checking the init datas, if we can't find a corresponding init data, we get the current session if there is only one session in the CDM instance with the corresponding `FIXME` comment.

Reply:
Sure we will change it. actually in the first commit it was in that order and we have changed it only based on your suggestion